platform/uwp/export/export.cpp

```diff
@@ -519,7 +519,9 @@ Error AppxPackager::add_file(String p_file_name, const uint8_t *p_buffer, size_t
 
 			int total_out_before = strm.total_out;
 
-			deflate(&strm, Z_FULL_FLUSH);
+			int err = deflate(&strm, Z_FULL_FLUSH);
+			ERR_FAIL_COND_V(Error(err) != OK, ERR_BUG);
```

Comment:
We shouldn't cast thirdparty error codes to our own, there's no guarantee that we use the same conventions. Here the equivalent woudl be `err != Z_OK` (see `zlib.h`), though I think `Z_STREAM_END` (1) is also a success return value:
```
deflate() returns Z_OK if some progress has been made (more input
  processed or more output produced), Z_STREAM_END if all input has been
  consumed and all output has been produced (only when flush is set to
  Z_FINISH), Z_STREAM_ERROR if the stream state was inconsistent (for example
  if next_in or next_out was Z_NULL or the state was inadvertently written over
  by the application), or Z_BUF_ERROR if no progress is possible (for example
  avail_in or avail_out was zero).  Note that Z_BUF_ERROR is not fatal, and
  deflate() can be called again with more input and more output space to
  continue compressing.
```

Maybe `err >= 0` would be safer as per:
```
#define Z_OK            0
#define Z_STREAM_END    1
#define Z_NEED_DICT     2
#define Z_ERRNO        (-1)
#define Z_STREAM_ERROR (-2)
#define Z_DATA_ERROR   (-3)
#define Z_MEM_ERROR    (-4)
#define Z_BUF_ERROR    (-5)
#define Z_VERSION_ERROR (-6)
/* Return codes for the compression/decompression functions. Negative values
 * are errors, positive values are used for special but normal events.
 */
```